src/apify/scrapy/utils.py

```diff
@@ -0,0 +1,168 @@
+import asyncio
+import codecs
+import pickle
+import random
+import string
+
+try:
+    from scrapy import Request, Spider
+    from scrapy.utils.request import request_from_dict
+except ImportError as exc:
+    raise ImportError(
+        'To use this module, you need to install the "scrapy" extra. Run "pip install apify[scrapy]".',
+    ) from exc
+
+from ..actor import Actor
+from ..storages import RequestQueue, StorageClientManager
+
+nested_event_loop: asyncio.AbstractEventLoop = asyncio.new_event_loop()
+
+
+def get_random_id(length: int = 6) -> str:
```

Comment:
How about using `_crypto_random_object_id`? I'm not sure we need another method for ids

Reply:
By `_crypto_random_object_id` you mean any function specifically?

Otherwise e.g. [secrets.token_urlsafe](https://docs.python.org/3/library/secrets.html#secrets.token_urlsafe) funcion could be used, it generates alphanumeric (+ underscore) strings, which is fine.

Reply:
there's a method for it in the sdk or shared

Reply:
Ahh... nice, thanks.